src/init.cpp

```diff
@@ -1713,8 +1699,16 @@ bool AppInitMain(InitInterfaces& interfaces)
 #endif
 
     std::vector<fs::path> vImportFiles;
-    for (const std::string& strFile : gArgs.GetArgs("-loadblock")) {
-        vImportFiles.push_back(strFile);
+
+    if (gArgs.IsArgSet("-loadblock")) {
+        const std::vector<std::string> vLoadBlockArgs = gArgs.GetArgs("-loadblock");
+
+        // Regard the '-noloadblock' argument
+        if (!vLoadBlockArgs.empty()) {
```

Comment:
You could remove this check, iterating an empty array is fine.

Reply:
done